msprime/provenance.py

```diff
@@ -150,8 +148,6 @@ def hook(obj):
                 return types.FunctionType(
                     marshal.loads(base64.b64decode(obj['__function__'])),
                     {}, 'func', obj['defaults'], ())
-            elif '__ts__' in obj:
-                return tskit.TableCollection.fromdict(obj['__ts__']).tree_sequence()
```

Comment:
Similarly here instead of removing replace with something like:
```python
elif '__constant__' in obj:
    return {
        '__current_ts__': tskit.CURRENT_TREE_SEQUENCE
    }[obj['__constant__']
```